src/containers/RecordingsTable.tsx

```diff
@@ -54,40 +52,32 @@ const RecordingsTable: FunctionComponent<Props> = ({ recordings, sortings, onDel
 
     recordings = sortByKey(recordings, 'recordingLabel');
 
-    const effect = async () => {
-        recordings.forEach(rec => {
-            (async () => {
-                if ((!rec.recordingInfo) && (!rec.fetchingRecordingInfo)) {
-                    // todo: use calculationPool for this
-                    rec.fetchingRecordingInfo = true;
-                    try {
-                        const info = await getRecordingInfo({recordingObject: rec.recordingObject, hither});
-                        onSetRecordingInfo({ recordingId: rec.recordingId, recordingInfo: info });
-                    }
-                    catch (err) {
-                        console.error(err);
-                        return;
-                    }
-                }
-            })();
-        });
-    }
-    useEffect(() => { effect() })
+    const handleViewRecording = useCallback((recording: Recording) => {
+        workspaceRouteDispatch({
+            type: 'gotoRecordingPage',
+            recordingId: recording.recordingId
+        })
+    }, [workspaceRouteDispatch])
+
+    const recordingInfos: {[key: string]: RecordingInfo} = useRecordingInfos(recordings)
```

Comment:
Here's a hook for retrieving the recordingInfo for a list of recordings. This hook simplifies things a lot, as you can see by the deleted lines of code.